wallet/src/de/schildbach/wallet/payments/parsers/PaymentIntentParser.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2023 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.payments.parsers
+
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.data.PaymentIntent
+import de.schildbach.wallet.ui.util.InputParser
+import de.schildbach.wallet.util.AddressUtil
+import de.schildbach.wallet.util.Io
+import de.schildbach.wallet_test.R
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.bitcoinj.core.Address
+import org.bitcoinj.core.AddressFormatException
+import org.bitcoinj.core.Base58
+import org.bitcoinj.protocols.payments.PaymentProtocol
+import org.bitcoinj.protocols.payments.PaymentProtocolException
+import org.bitcoinj.protocols.payments.PaymentProtocolException.PkiVerificationException
+import org.bitcoinj.uri.BitcoinURI
+import org.bitcoinj.uri.BitcoinURIParseException
+import org.dash.wallet.common.util.Base43
+import org.dash.wallet.common.util.ResourceString
+import org.slf4j.LoggerFactory
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.util.*
+import java.util.regex.Pattern
+
+class PaymentIntentParserException(
+    innerException: Exception,
+    val localizedMessage: ResourceString
+) : Exception(innerException)
+
+object PaymentIntentParser {
```

Comment:
`InputParser` is hard to use from Kotlin and is responsible for too many things. Most callers know which parsing functionality they need and do not require all of the callbacks `InputParser` has. 

I think it's better to have several dedicated parsing classes instead.